reference/dev-activity-terminology.md

```diff
@@ -0,0 +1,25 @@
+Development Activity Terminology
+================================
+
+We have several mature software products. The terminology around maintenance and development can be ambiguous. This doc attempts to define some terms so that we all know what we’re talking about. 
+
+## Maintenance 
+
+The work required to maintain the current level of functionality, while continuing to protect against changing threats. Usually this is the work of applying updates to upstream libraries, upgrading away from end-of-life versions of software, making decisions about what to do when functionality in software changes.
+
+## Bugfixes
+
+Correcting the software when a problem has been found that means that the software does not work as designed, or to correct a design decision that doesn’t achieve the desired functionality. For example, this might be correcting a typo, or fixing a function that doesn’t do what it says.  
+
+## Hosting 
+
+For services that deliver functionality over the Web (as opposed to software that’s downloaded and run on someone’s own computer), hosting is the provision of server space, and the work to assure the regular availability of that server space - such as monitoring, performing routine software upgrades, and responding to incidents. 
+
+## Incremental Improvements 
+
+Changes to the software that enhance its functionality that don’t require significant changes to the general approach of delivering the functionality of the software. For example, this might be adding a new data check or type of export, adding a new type of functionality to an existing tool, or changing the presentation of some results. 
+
+## Iterative Improvements 
+
+Larger pieces of work to make a step-change to a product, that may change the general approach to delivering the functionality of the software. For example - a redesign of an interface, a behind-the-scenes reworking of functionality to be faster or handle more data, the splitting up or combining of existing products to better serve users’ needs.
+
```

Comment:
To me Iterative and Incremental are too similar, you can increment and iterate by the same amount of change. What about Minor and Major improvements?

Reply:
I think that the two types of work here are conceptually quite different - it's not just a question of scale. I'm not precious about the words, but I think that the conceptual difference is important. 

One type of change - what I've called an Iterative one - is a whole trip around the cycle of assessing needs, exploring options, developing and validating a solution, and putting it into use. Even if it has quite a tight scope (which is often a good thing!), it still requires us to apply some research and design thinking to the work.

Another type of change - what I've called Incremental - is a much lighter process, and is more applicable where there isn't any real doubt about how something should work. Often, that's changes that might not count as bugfixes (because they're not strictly *wrong* or *broken*) but that are of relatively little consequence and don't change our overall approach. 

The scale of the changes may well overlap, but I don't think this is really about scale. 

Does that conceptual separation make sense? Can you suggest a more helpful name for it?